pyvrp/GeneticAlgorithm.py

```diff
@@ -117,43 +117,45 @@ def run(self, stop: StoppingCriterion):
         iters_no_improvement = 1
 
         for individual in self._initial_solutions:
-            self._pop.add(individual)
+            self._pop.add(individual, self._pm)
 
-        while not stop(self._best):
+        while not stop(self._best.cost(self._pm)):
             iters += 1
 
             if iters_no_improvement == self._params.nb_iter_no_improvement:
                 iters_no_improvement = 1
                 self._pop.clear()
 
                 for individual in self._initial_solutions:
-                    self._pop.add(individual)
+                    self._pop.add(individual, self._pm)
 
-            curr_best = self._best.cost()
+            curr_best = self._best.cost(self._pm)
 
             parents = self._pop.select(self._rng)
             offspring = self._op(parents, self._data, self._pm, self._rng)
             self._educate(offspring)
 
-            new_best = self._best.cost()
+            new_best = self._best.cost(self._pm)
 
             if new_best < curr_best:
                 iters_no_improvement = 1
             else:
                 iters_no_improvement += 1
 
             if self._params.collect_statistics:
-                stats.collect_from(self._pop)
+                stats.collect_from(self._pop, self._pm)
 
         end = time.perf_counter() - start
         return Result(self._best, stats, iters, end)
 
     def _educate(self, individual: Individual):
         def is_new_best(indiv):
-            return indiv.is_feasible() and indiv.cost() < self._best.cost()
+            return indiv.is_feasible() and indiv.cost(
+                self._pm
+            ) < self._best.cost(self._pm)
```

Comment:
Can we assign the indiv's cost to a variable before doing this check? I think that's much easier to read over two lines than what we have currently.

Something like:
```python
cost = indiv.cost(self._pm)
return indiv.is_feasible() and cost < self._best.cost(self._pm)
```

Reply:
I agree it's not pretty (did not check the auto formatting). However it seems arbitrary to me to assign `cost` to a variable but have `best_cost` computed inline, so I propose either:
```
cost = indiv.cost(self._pm)
best_cost = self._best.cost(self._pm)
return indiv.is_feasible() and cost < best_cost
```
or
```
return (
    indiv.is_feasible()
    and indiv.cost(self._pm) < self._best.cost(self._pm)
)
```
I prefer declarative over imperative so I choose the second option. Let me know if you have strong objections.

Reply:
Nevermind, apparently black doesn't allow me the second option so I implemented the first.